examples/js/DragNDrop.js

```diff
@@ -0,0 +1,173 @@
+/* global itowns */
+var DragNDrop = (function _() {
+    // TYPE
+    var _TEXT = 1;
+    var _JSON = 2;
+    var _BINARY = 3;
+    var _IMAGE = 4;
+    var _XML = 5;
+
+    // MODE
+    var _COLOR = 6;
+    var _GEOMETRY = 7;
+
+    var extensionsMap = {};
+    var fileReader = new FileReader();
+
+    var _view;
+
+    function addFiles(event, files) {
+        event.preventDefault();
+
+        // Read each file
+        for (var i = 0; i < files.length; i++) {
+            var file = files[i];
+            var extension = extensionsMap[file.name.split('.').pop()];
+
+            // eslint-disable-next-line no-loop-func
+            fileReader.onload = function onload(e) {
+                var data = e.target.result;
+
+                if (extension.type == _JSON) {
+                    data = JSON.parse(data);
+                } else if (extension.type == _XML) {
+                    data = new window.DOMParser().parseFromString(data, 'text/xml');
+                }
+
+                extension.parser(data, {
+                    buildExtent: true,
+                    crsIn: 'EPSG:4326',
```

Comment:
The projection doesn't work for all cases.

Reply:
True, I don't know how to read the projection of each file. Does it needs to be added as well when registering a format ?

Reply:
No, but could you add this comment in the documentation and add a message in a `div` in your html example.